<a href="https://colab.research.google.com/github/Vivek-Hiremath55/AI_Systems_Pro/blob/main/VivekHiremath/Playground_VivekHiremath.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Initializing important libraries

In [2]:
import pandas as pd, matplotlib.pyplot as plt

# Initializing Descriptive Dataset

In [5]:
train_meta_df = pd.read_csv('train-metadata.csv')

<ipython-input-5-2dcdef161a58>:1: DtypeWarning: Columns (51,52) have mixed types. Specify dtype option on import or set low_memory=False.
  train_meta_df = pd.read_csv('train-metadata.csv')


# Visualizing Dataframe

In [6]:
train_meta_df.head()

isic_id  target  patient_id  age_approx   sex anatom_site_general  \
0  ISIC_0015670       0  IP_1235828        60.0  male     lower extremity   
1  ISIC_0015845       0  IP_8170065        60.0  male           head/neck   
2  ISIC_0015864       0  IP_6724798        60.0  male     posterior torso   
3  ISIC_0015902       0  IP_4111386        65.0  male      anterior torso   
4  ISIC_0024200       0  IP_8313778        55.0  male      anterior torso   

   clin_size_long_diam_mm          image_type tbp_tile_type   tbp_lv_A  ...  \
0                    3.04  TBP tile: close-up     3D: white  20.244422  ...   
1                    1.10  TBP tile: close-up     3D: white  31.712570  ...   
2                    3.40  TBP tile: close-up        3D: XP  22.575830  ...   
3                    3.22  TBP tile: close-up        3D: XP  14.242329  ...   
4                    2.73  TBP tile: close-up     3D: white  24.725520  ...   

    lesion_id  iddx_full  iddx_1  iddx_2  iddx_3  iddx_4  iddx_5  \
0         NaN     Benign  Benign     NaN     NaN     NaN     NaN   
1  IL_6727506     Benign  Benign     NaN     NaN     NaN     NaN   
2         NaN     Benign  Benign     NaN     NaN     NaN     NaN   
3         NaN     Benign  Benign     NaN     NaN     NaN     NaN   
4         NaN     Benign  Benign     NaN     NaN     NaN     NaN   

   mel_mitotic_index  mel_thick_mm  tbp_lv_dnn_lesion_confidence  
0                NaN           NaN                     97.517282  
1                NaN           NaN                      3.141455  
2                NaN           NaN                     99.804040  
3                NaN           NaN                     99.989998  
4                NaN           NaN                     70.442510  

[5 rows x 55 columns]

# Checking for null values in df

In [7]:
train_meta_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 401059 entries, 0 to 401058
Data columns (total 55 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   isic_id                       401059 non-null  object 
 1   target                        401059 non-null  int64  
 2   patient_id                    401059 non-null  object 
 3   age_approx                    398261 non-null  float64
 4   sex                           389542 non-null  object 
 5   anatom_site_general           395303 non-null  object 
 6   clin_size_long_diam_mm        401059 non-null  float64
 7   image_type                    401059 non-null  object 
 8   tbp_tile_type                 401059 non-null  object 
 9   tbp_lv_A                      401059 non-null  float64
 10  tbp_lv_Aext                   401059 non-null  float64
 11  tbp_lv_B                      401059 non-null  float64
 12  tbp_lv_Bext                   401059 non-nul

In [14]:
train_meta_df.shape

(401059, 55)

# ------------------------------------------------------------------------------------------------------------------------------------------

### Number of Features = 54
### Data Type = Varied (object, float64, int64)
### Shape = Rows -> 401059, Columns -> 55

# ------------------------------------------------------------------------------------------------------------------------------------------

# Evaluating percentage of missing vallues

In [11]:
null_list = []

for col in train_meta_df.columns:
    pct_missing = train_meta_df[col].isna().sum()
    null_list.append(pct_missing * 100 / len(train_meta_df))

null = pd.DataFrame({'Column': train_meta_df.columns, 'Null Count': null_list})

In [13]:
null.sort_values(by='Null Count', ascending=False)

Column  Null Count
51                        iddx_5   99.999751
52             mel_mitotic_index   99.986785
53                  mel_thick_mm   99.984292
50                        iddx_4   99.862614
49                        iddx_3   99.734453
48                        iddx_2   99.733705
45                     lesion_id   94.500061
4                            sex    2.871647
5            anatom_site_general    1.435200
3                     age_approx    0.697653
33             tbp_lv_norm_color    0.000000
36                   tbp_lv_stdL    0.000000
35   tbp_lv_radial_color_std_max    0.000000
34            tbp_lv_perimeterMM    0.000000
0                        isic_id    0.000000
32            tbp_lv_norm_border    0.000000
31        tbp_lv_nevi_confidence    0.000000
38             tbp_lv_symm_2axis    0.000000
37                tbp_lv_stdLExt    0.000000
44             copyright_license    0.000000
39       tbp_lv_symm_2axis_angle    0.000000
40                      tbp_lv_x    0.000000
41                      tbp_lv_y    0.000000
42                      tbp_lv_z    0.000000
43                   attribution    0.000000
29        tbp_lv_location_simple    0.000000
46                     iddx_full    0.000000
47                        iddx_1    0.000000
30            tbp_lv_minorAxisMM    0.000000
27           tbp_lv_eccentricity    0.000000
28               tbp_lv_location    0.000000
15                      tbp_lv_H    0.000000
2                     patient_id    0.000000
6         clin_size_long_diam_mm    0.000000
7                     image_type    0.000000
8                  tbp_tile_type    0.000000
9                       tbp_lv_A    0.000000
10                   tbp_lv_Aext    0.000000
11                      tbp_lv_B    0.000000
12                   tbp_lv_Bext    0.000000
13                      tbp_lv_C    0.000000
14                   tbp_lv_Cext    0.000000
16                   tbp_lv_Hext    0.000000
1                         target    0.000000
17                      tbp_lv_L    0.000000
18                   tbp_lv_Lext    0.000000
19                tbp_lv_areaMM2    0.000000
20       tbp_lv_area_perim_ratio    0.000000
21         tbp_lv_color_std_mean    0.000000
22                 tbp_lv_deltaA    0.000000
23                 tbp_lv_deltaB    0.000000
24                 tbp_lv_deltaL    0.000000
25                tbp_lv_deltaLB    0.000000
26            tbp_lv_deltaLBnorm    0.000000
54  tbp_lv_dnn_lesion_confidence    0.000000